src/library/mixin.lua

```diff
@@ -4,227 +4,368 @@
 --[[
 $module Fluid Mixins
 
-The Fluid Mixins library simplifies the process of writing plugins and improves code maintainability.
+The Fluid Mixins library does the following:
+- Modifies Finale objects to allow methods to be overridden and new methods or properties to be added. In other words, the modified Finale objects function more like regular Lua tables.
+- Mixins can be used to address bugs, to introduce time-savers, or to provide custom functionality.
+- Introduces a new namespace for accessing the mixin-enabled Finale objects.
+- Also introduces two types of formally defined mixin: `FCM` and `FCX` classes
+- As an added convenience, all methods that return zero values have a fluid interface enabled (aka method chaining)
+
+
+## finalemix Namespace
+To utilise the new namespace, simply include the library, which also gives access to he helper functions:
+```lua
+local finalemix = require('library.mixin')
+```
 
-This library does 2 things:
-- Allows mixins to be added to any `FC*` objects
-- Adds a fluid interface for any methods that return zero values.
+All defined mixins can be accessed through the `finalemix` namespace in the same way as the `finale` namespace. All constructors have the same signature as their `FC` originals.
 
+```lua
+local fcstr = finale.FCString()
 
-By default, it is not possible to override or extend any of the `FC*` objects in Finale Lua because objects that are tied to their underlying C++ implementation are userdata, not tables.
-So through a little bit of magic with proxying, this library enables new methods and properties to be added, while maintaining access to the original methods and properties.
+-- Base mixin-enabled FCString object
+local fcmstr = finalemix.FCMString()
 
+-- Customised mixin that extends FCMString
+local fcxstr = finalemix.FCXString()
 
-**Adding New Methods and Properties**
-Including the mixin library instantly enables the addition of new methods and properties. The example below demonstrates adding a new property and a new method which accesses that property.
+-- Customised mixin that extends FCXString. Still has the same constructor signature as FCString
+local fcxstr2 = finale.FCXMyCustomString()

```

Comment:
Is `finale` correct here? Did you mean `finalemix` or `fcsstr` instead?

Reply:
Good catch, fixed.